src/pbft/pbft_config.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#include "libbyz/libbyz.h"
+#include "libbyz/pbft_assert.h"
+#include "pbft/pbft_deps.h"
+
+namespace pbft
+{
+  static char* service_mem = 0;
+
+  class PbftConfig
+  {
+  public:
+    virtual ~PbftConfig() = default;
+    virtual ExecCommand get_exec_command() = 0;
+    virtual size_t message_size() = 0;
+    virtual void fill_request(
+      uint8_t* buffer,
+      const size_t& total_req_size,
+      const std::vector<uint8_t>& data,
+      const size_t& version) = 0;
+  };
+
+  class PbftConfigCCF : public PbftConfig
+  {
+  public:
+    ~PbftConfigCCF() = default;
+
+    size_t message_size() override
+    {
+      return sizeof(ccf_req);
+    }
+
+    void fill_request(
+      uint8_t* buffer,
+      const size_t& total_req_size,
+      const std::vector<uint8_t>& data,
+      const size_t& version) override
+    {
+      auto request = new (buffer) ccf_req;
+      request->version = version;
+      auto array_size = request->get_array_size(total_req_size);
+
+      for (size_t j = 0; j < array_size; j++)
+      {
+        memcpy(&request->get_counter_array()[j], &data[j], sizeof(uint8_t));
+      }
+    }
+
+    struct ccf_req
+    {
+      size_t version;
+
+      uint8_t* get_counter_array()
+      {
+        return (uint8_t*)((uintptr_t)this + sizeof(ccf_req));
+      }
+
+      size_t get_array_size(size_t total_size)
+      {
+        if (total_size < sizeof(ccf_req))
+        {
+          return 0;
+        }
+        return (total_size - sizeof(ccf_req)) / sizeof(uint8_t);
+      }
+    };
+
+  private:
+    ExecCommand exec_command = [](
+                                 Byz_req* inb,
+                                 Byz_rep* outb,
+                                 _Byz_buffer* non_det,
+                                 int client,
+                                 bool ro,
+                                 Seqno total_requests_executed) {
+      Long& counter = *(Long*)service_mem;
```

Comment:
Could we write a `TODO` describing what `exec_command` is currently doing (inspired by the pbft counter example and nothing to do with CCF for now) and explain that this will change soon?